### PR TITLE
fix: catch edge case of lang in path

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Handles the edgiest of edge cases where paths include the current or target client lang as a path segment: e.g. espanol/learn/responsive-web-design/basic-html-and-html5/espanol

### CODE REVIEW

File: client/src/components/createLanguageRedirect.js [hunk 1]
```diff
@@ -4,10 +4,12 @@ const createLanguageRedirect = ({ clientLocale, lang }) => {
...
   let path = window?.location?.pathname?.split('/');
   path = path
-    .filter(item => (item !== clientLocale && item !== lang ? item : ''))
+    .filter(
+      (item, i) => i !== 0 || (item !== clientLocale && item !== lang && item)
+    )
     .join('/');
 
-  const domain = window?.location?.host
+  const domain = window?.location?.hostname
```
Comment (raisedadead) [inline client/src/components/createLanguageRedirect.js:12]: Any reason we would not want `host` (not that we need it, but is inclusive of port and can be used in future for tests, elsewhere?)

File: client/src/components/createLanguageRedirect.test.js [hunk 2]
```diff
@@ -59,6 +64,16 @@ describe('createLanguageRedirect for clientLocale === english', () => {
       });
       expect(receivedPageURL).toBe(francaisPageURL);
     });
+
+    it('should not replace the second espanol', () => {
```
Comment (raisedadead) [inline client/src/components/createLanguageRedirect.test.js:68]: If we have ever had such a use case, then we would need to rethink it instead.

File: client/src/components/createLanguageRedirect.js (hunk 1 above)
Comment (ojeytonwilliams) [reply client/src/components/createLanguageRedirect.js:12]: I guess not. I just saw the name `domain` and thought that would mean we didn't want the port.

Maybe hostTail or some other slightly awkward, but informative, name would work better.

File: client/src/components/createLanguageRedirect.test.js (hunk 2 above)
Comment (ojeytonwilliams) [reply client/src/components/createLanguageRedirect.test.js:68]: Agreed. I get that this is an edge case, but I figured it was easy to prevent.

Comment (raisedadead) [reply client/src/components/createLanguageRedirect.test.js:68]: I think this change needs to be reverted, because we know we do not have such a case as of now. When we do have a language segment in the path we need to think the "why" we have the case. 

It needs be caught (most likely by a user) as eitherways (prevent it now or not). I would much rather not fix it yet.

File: client/src/components/createLanguageRedirect.js [hunk 3]
```diff
@@ -4,15 +4,17 @@ const createLanguageRedirect = ({ clientLocale, lang }) => {
 
   let path = window?.location?.pathname?.split('/');
   path = path
-    .filter(item => (item !== clientLocale && item !== lang ? item : ''))
+    .filter(
+      (item, i) => i !== 0 || (item !== clientLocale && item !== lang && item)
+    )
```
Comment (raisedadead) [inline client/src/components/createLanguageRedirect.js:9]:
```suggestion
    .filter(item => (item !== clientLocale && item !== lang ? item : ''))
```

File: client/src/components/createLanguageRedirect.test.js [hunk 4]
```diff
@@ -16,6 +16,11 @@ describe('createLanguageRedirect for clientLocale === english', () => {
       'https://www.freecodecamp.org/espanol/learn/responsive-web-design/basic-html-and-html5/inform-with-the-paragraph-element';
     const francaisPageURL =
       'https://www.freecodecamp.org/francais/learn/responsive-web-design/basic-html-and-html5/inform-with-the-paragraph-element';
+    const singleEspanolURL =
+      'https://www.freecodecamp.org/learn/responsive-web-design/basic-html-and-html5/espanol';
+
+    const doubleEspanolURL =
+      'https://www.freecodecamp.org/espanol/learn/responsive-web-design/basic-html-and-html5/espanol';
```
Comment (raisedadead) [inline client/src/components/createLanguageRedirect.test.js:23]:
```suggestion
```

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: client/src/components/createLanguageRedirect.test.js [hunk 5]
```diff
@@ -59,6 +59,16 @@ describe('createLanguageRedirect for clientLocale === english', () => {
...
+
+    it('should not replace the second espanol', () => {
+      delete window.location;
+      window.location = new URL(singleEspanolURL);
+      const receivedPageURL = createLanguageRedirect({
+        ...envVars,
+        lang: 'espanol'
+      });
+      expect(receivedPageURL).toBe(doubleEspanolURL);
+    });
```
Comment (raisedadead) [inline client/src/components/createLanguageRedirect.test.js:71]:
```suggestion
```

File: client/src/components/createLanguageRedirect.test.js [hunk 6]
```diff
@@ -59,6 +59,7 @@ describe('createLanguageRedirect for clientLocale === english', () => {
       });
       expect(receivedPageURL).toBe(francaisPageURL);
     });
+
```
Comment (raisedadead) [inline client/src/components/createLanguageRedirect.test.js:62]:
```suggestion
```
